docs/src/architecture/notes.md

```diff
@@ -11,54 +14,62 @@ In the above:
 * **Inputs** which are placed onto the stack before a note's script is executed.
 * **Serial number** is the note's identifier.
 
-## Vault
-Asset container for a note. A note vault can contain up to 255 assets. The entire vault can be reduced to a single hash which is computed by sequentially hashing the list of the vault's assets.
+### Vault
+Asset container for a note. A note vault can contain up to `255` assets stored in an array. The entire vault can be reduced to a single hash which is computed by sequentially hashing the list of the vault's assets.
 
-A note's vault is basically the same as an account's vault.
+### Script
+Unlike an account, a note has a single executable script. This script will be executed in a [transaction](https://0xpolygonmiden.github.io/miden-base/architecture/transactions.html). This script is also the root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html). A script is always executed in the context of a single account, and thus, may invoke account's functions. A note's script can call zero or more of an account's function.
 
-## Script
-Unlike an account, a note has a single executable script. This script is also a root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html). A script is always executed in the context of a single account, and thus, may invoke account's functions. A note script does not have to call any of account's functions. More generally, a note's script can call zero or more of an account's function.
+*Note: Since code in Miden is expresed as MAST, every function is a commitment to the underlying code. The code cannot change unnoticed to the user because its hash would change. Behind any MAST root there can only be `256` functions*
```

Comment:
"Behind any MAST root there can only be `256` functions*" I don't think this is accurate.  An account can only have 256 procedures but that isn't relevant to the note script.

Reply:
changed it. So there can be more than `256` functions behind a MAST root?